sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala

```diff
@@ -87,8 +87,22 @@ case class CustomShuffleReaderExec private(
     Iterator(desc)
   }
 
-  def hasCoalescedPartition: Boolean =
-    partitionSpecs.exists(_.isInstanceOf[CoalescedPartitionSpec])
+  private def isCoalesced(spec: ShufflePartitionSpec) = coalesceRange(spec) > 1
+
+  /**
+   * How many partitions were coalesced; 0 if not [[CoalescedPartitionSpec]]
+   */
+  private def coalesceRange(spec: ShufflePartitionSpec) = spec match {
+    case s: CoalescedPartitionSpec => s.endReducerIndex - s.startReducerIndex
+    case _ => 0
+  }
+
+  /**
+   * Returns true iff some non-empty partitions were combined
+   */
+  def hasCoalescedPartition: Boolean = {
+    partitionSpecs.exists(isCoalesced)
```

Comment:
nit:
```
partitionSpecs.exists {
  case s: CoalescedPartitionSpec => s.endReducerIndex - s.startReducerIndex
  case _ => false
}
```

Reply:
Can we use this simple implementation? 3 methods looks a bit overkill.

Reply:
Could we please agree first on the substance of this PR?  Specifically the answer to https://github.com/apache/spark/pull/32872/files#r649658312?
If that makes sense I can make the other changes.